File: Java/3_longest_common_prefix.java
import java.util.Arrays;

class LongestCommonPrefix {
    static void main(String[] args) {
        System.out.println(longestPrefix(new String[] {"flower","flow","flight"}));
    }

    static String longestPrefix(String [] array){
        Arrays.sort(array);
        int limit = Math.min(array[0].length(),array[array.length-1].length());
        for (int i = 0; i < limit; i++){
            if (array[0].charAt(i)!=array[array.length-1].charAt(i)){
                return array[0].substring(0,i);
            }
        }
        return array[0].substring(0,limit);
    }
}

File: TypeScript/3_longest_common_prefix.ts
export function longestCommonPrefix(strs: string[]): string {
  if (strs.length === 0) return "";
  const sorted = [...strs].sort();
  const first = sorted[0];
  const last = sorted[sorted.length - 1];
  const limit = Math.min(first.length, last.length);
  for (let i = 0; i < limit; i++) {
    if (first[i] !== last[i]) {
      return first.slice(0, i);
    }
  }
  return first.slice(0, limit);
}

console.log(longestCommonPrefix(["flower", "flow", "floweight"])); 